docs/integration/driver/wayland.rst

```diff
@@ -0,0 +1,182 @@
+=============================
+Wayland Display/Inputs driver
+=============================
+
+Overview
+--------
+
+| The **Wayland** `driver <https://github.com/lvgl/lvgl/tree/master/src/drivers/wayland>`__ offers support for simulating the LVGL display and keyboard/mouse inputs in a desktop window.
+| It is an alternative to **X11** or **SDL2**
+
+The main purpose for this driver is for testing/debugging the LVGL application, it can also be used to run applications in 'kiosk mode'
+
+Dependencies
+------------
+
+The wayland driver requires some dependencies.
+
+On Ubuntu
+
+.. code:: bash
+
+   sudo apt-get install libwayland-dev libxkbcommon-dev libwayland-bin wayland-protocols
+
+On Fedora
+
+.. code:: bash
+
+   sudo dnf install wayland-devel libxkbcommon-devel wayland-utils wayland-protocols-devel
+
+
+Configuring the wayland driver
+------------------------------
+
+1. Enable the wayland driver in ``lv_conf.h``
+
+.. code:: c
+
+   #define LV_USE_WAYLAND  1
+
+2. Optional configuration options:
+
+- Enable window decorations, only required on GNOME because out of all the available wayland compositors
+  **only** Mutter/GNOME enforces the use of client side decorations
+
+.. code:: c
+
+   #define LV_WAYLAND_WINDOW_DECORATIONS 1
+
+- Enable support for the deprecated 'wl_shell', Only useful when the BSP on the target has weston ``9.x``
+
+.. code:: c
+
+   #define LV_WAYLAND_WL_SHELL  1 /*preferred default*/
+
+Example
+-------
+
+An example simulator is available in this `repo <https://github.com/lvgl/lv_port_linux/>`__
+
+Usage
+-----
+
+#. In ``main.c`` ``#incude "lv_drivers/wayland/wayland.h"``
+#. Enable the Wayland driver in ``lv_conf.h`` with ``LV_USE_WAYLAND 1``
+
+#. ``LV_COLOR_DEPTH`` should be set either to ``32`` or ``16`` in ``lv_conf.h``
+
+#. After ``lv_init()`` call ``lv_wayland_init()``.
```

Comment:
Would it make sense to have `lv_wayland_init()` happen in lv_init.c when `LV_USE_WAYLAND` is `1`?

Reply:
Thanks! For consistency with the new patterns could you please move the function prototypes for `lv_wayland_init` and `lv_wayland_deinit` to a new file `lv_wayland_private.h` (and include that header in `lv_init.c` and `lv_wayland.c`)? Either that, or perform the init on the first call to `lv_wayland_window_create` if you prefer.

Reply:
@liamHowatt Ok, moved init/deinit declarations to a private header

Reply:
Calling the `lv_wayland_init()` function directly in the `lv_init()` function removes the ability to compile lvgl with multiple drivers enabled (e.g. sdl, fbdev, drm and wayland) in the config file:
```
#define LV_USE_SDL 1
#define LV_USE_LINUX_FBDEV 1
#define LV_USE_LINUX_DRM 1
#define LV_USE_WAYLAND 1
```
And then let the application choose the driver by calling the corresponding function to create the window (e.g. `lv_sdl_window_create`, `lv_linux_fbdev_create`, `lv_linux_drm_create` or `lv_wayland_window_create`).

This can be avoided by either requiring the developer to the call the initialization function manually again, or letting the window creation function take care of calling it internally.

Reply:
The way it is now looks good. I think maybe my suggestion to do it it lv_init.c was bad advice.

Thanks!

Just a question: it says multiple window support is experimental for now. Do you know what happens if a second window is created? Will it re-initialize Wayland and put the first window into an erroneous state?